src/pymatgen/symmetry/settings.py

```diff
@@ -100,6 +98,10 @@ def parse_transformation_string(
         Returns:
             tuple[list[list[float]] | np.ndarray, list[float]]: transformation matrix & vector
         """
+        # Import sympy is expensive
```

Comment:
## import profile of `core.structure`

### Current master branch with `torch`

<img width="1505" alt="image" src="https://github.com/user-attachments/assets/7c21db99-53a8-4050-ad00-90a8d31d586f">


### After `monty.json` patch

<img width="1494" alt="image" src="https://github.com/user-attachments/assets/f77ad6e3-90f2-48d4-bd21-d2102a3fb451">


### After lazily load `sympy`

`sympy` is required by `symmetry.settings.JonesFaithfulTransformation.parse_transformation_string`, and reduce import time by about 15%, sounds like a good deal

<img width="1494" alt="image" src="https://github.com/user-attachments/assets/612a2b4c-ed45-41ea-9f3c-03e0c68e2083">


### After lazily load `scipy.spatial.Voronoi` (reverted)

Usage of `scipy` seem very common across `pymatgen` too, doesn't bring much benefit, reverted.